akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -1177,18 +1182,26 @@ private[akka] class ActorSystemImpl(
      * @param throwOnLoadFail Throw exception when an extension fails to load (needed for backwards compatibility)
      */
     def loadExtensions(key: String, throwOnLoadFail: Boolean): Unit = {
+      def handleException(fqcn: String, problem: Throwable): Unit = {
+        if (!throwOnLoadFail) log.error(problem, "While trying to load extension [{}], skipping...", fqcn)
+        else throw new RuntimeException(s"While trying to load extension [$fqcn]", problem)
+      }
+
       immutableSeq(settings.config.getStringList(key)).foreach { fqcn =>
         dynamicAccess.getObjectFor[AnyRef](fqcn).recoverWith {
           case _ => dynamicAccess.createInstanceFor[AnyRef](fqcn, Nil)
         } match {
-          case Success(p: ExtensionIdProvider) => registerExtension(p.lookup())
-          case Success(p: ExtensionId[_])      => registerExtension(p)
+          case Success(p: ExtensionIdProvider) =>
+            try registerExtension(p.lookup())
+            catch { case NonFatal(e) => handleException(fqcn, e) }
+          case Success(p: ExtensionId[_]) =>
+            try registerExtension(p)
+            catch { case NonFatal(e) => handleException(fqcn, e) }
```

Comment:
using the same error handling here as when the fqcn couldn't be loaded. Not necessary for fixing the test failure but seems more consistent.